src/libDirectoryService/FinalBlockPostProcessing.cpp

```diff
@@ -180,11 +181,17 @@ void DirectoryService::ProcessFinalBlockConsensusWhenDone() {
     t_microBlocks.emplace(microBlock.GetHeader().GetShardId(), microBlock);
   }
 
+  DequeOfShard t_shards;
+  if (m_forceMulticast && GUARD_MODE) {
+    ReloadGuardedShards(t_shards);
+  }
+
   DataSender::GetInstance().SendDataToOthers(
-      *m_finalBlock, *m_mediator.m_DSCommittee, m_shards, t_microBlocks,
+      *m_finalBlock, *m_mediator.m_DSCommittee,
+      (m_forceMulticast && GUARD_MODE) ? t_shards : m_shards, t_microBlocks,
```

Comment:
Maybe can be changed like following, to avoid multiple comparison

   DequeOfShard t_shards;
  if (m_forceMulticast && GUARD_MODE) {
    ReloadGuardedShards(t_shards);
  }
  else {
    t_shards = m_shards;
  }

   DataSender::GetInstance().SendDataToOthers(
      *m_finalBlock, *m_mediator.m_DSCommittee, m_shards, t_microBlocks,